CoreScriptsRoot/Modules/DeveloperConsoleModule.lua

```diff
@@ -3015,4 +3024,46 @@ function DevConsoleModuleTable:SetVisibility(value)
 	devConsole:SetVisible(value)
 end
 
+
+local creatingLock = false
+local creatingVisibleValueToSet = false
+
+local function SetCoreConsoleCreation()
+	if (creatingLock) then return end
+	creatingLock = true
+
+	spawn(function()
+		--// Keep GetVisibility call before SetVisibility because the first call will yield for some time and 
+		--// there is the possibility that during the yield time the value of 'creatingVisibleValueToSet' may
+		--// change.
+		DevConsoleModuleTable:GetVisibility()
+		DevConsoleModuleTable:SetVisibility(creatingVisibleValueToSet)
+
+		creatingLock = false
+	end)
+end
+
+local StarterGui = game:GetService("StarterGui")
+StarterGui:RegisterGetCore("DeveloperConsoleVisible", function()
```

Comment:
The tests I ran had SetCore and GetCore calls immediately after a GetService("StarterGui") in a LocalScript running in StarterPlayerScripts and everything was fine. DeveloperConsoleModule.lua seems to be required at startup.